src/todo/todo.service.ts

```diff
@@ -1,29 +1,55 @@
-import { Injectable, NotFoundException } from '@nestjs/common';
+import {
+  HttpException,
+  HttpStatus,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
 import { TodoRepository } from './todo.repository';
-import { CreateTodoProps, DeleteTodoProps, GetTodoByIdProps } from './types';
+import {
+  CreateTodoProps,
+  DeleteTodoProps,
+  GetAllTodoProps,
+  GetTodoByIdProps,
+  MarkTodoAsCompletedProps,
+  Todo,
+} from './types';
 @Injectable()
 export class TodoService {
   constructor(private readonly _todoRepository: TodoRepository) {}
 
-  create(props: CreateTodoProps) {
+  create(props: CreateTodoProps): Promise<Todo> {
     return this._todoRepository.create(props);
   }
 
-  findAll(userId: string) {
-    return this._todoRepository.findAll(userId);
+  findAll(props: GetAllTodoProps): Promise<Todo[]> {
+    return this._todoRepository.findAll(props);
   }
 
-  async findOne(props: GetTodoByIdProps) {
-    const task = await this._todoRepository.findOne(props);
-    if (!task) {
+  async findOne(props: GetTodoByIdProps): Promise<Todo> {
+    const todo = await this._todoRepository.findOne(props);
```

Comment:
Should check with user id, unless user A can query user B todo data

Reply:
I think it isn't necessary because the props that I pass to the `todoRepository` contains `userId`, so the query condition has included this.